doc/hardware/clock_mgmt/index.rst

```diff
@@ -0,0 +1,338 @@
+Zephyr provides the clock management subsystem in order to enable clock
+consumers and applications to manage clocks in a device-agnostic manner.
+
+Reading Clocks
+**************
+
+Drivers must supply a subsystem identifier when reading or subscribing to
+a callback for a clock. The clock management defines one subsystem identifier,
+:c:macro:`CLOCK_MGMT_SUBSYS_DEFAULT`. Consumers may define additional private
+identifiers if they need to query multiple clocks. Each identifier corresponds
+to an index within the ``clocks`` property of the node defining the clock
+consumer.
+
+Consumers can query a clock output rate via the :c:func:`clock_mgmt_get_rate`
+function. This function returns the clock rate as a positive integer,
+or a negative error value on failure.
+
+Consumers can also monitor a clock output rate. To do so, they may
+register for a callback. Once a callback is initialized with
+:c:func:`clock_mgmt_init_callback`, it can be registered with the
+:c:func:`clock_mgmt_add_callback` function. Until the callback is removed
+with :c:func:`clock_mgmt_remove_callback`, the driver will receive a callback
+whenever the clock changes rate, starts, or stops.
+
+Setting Clocks
+**************
+
+Clocks are primarily managed via "setpoints". A setpoint corresponds to a
+specific group of clock settings, which may have systemwide effects, or simply
+configure clocks for a given peripheral. For example, one setpoint may simply
+configure the multiplexer and divider for a given peripheral. Another may
+reconfigure the system level PLL, and modify the core clock rate.
+
+Setpoints are configured via devicetree, and are usually opaque to the
+driver/application code consuming the clock. Instead of having direct
+knowledge of a setpoint's contents, clock consumers select a setpoint
+based on the power state they are targeting. The following setpoint
+power states are defined:
+
+.. table:: Standardized state names
+    :align: center
+
+    +-------------+-------------------------------------+-------------------------+
+    | State       | Identifier                          | Purpose                 |
+    +-------------+-------------------------------------+-------------------------+
+    | ``default`` | :c:macro:`CLOCK_MGMT_STATE_DEFAULT` | Clock state when        |
+    |             |                                     | the device is in        |
+    |             |                                     | operational state       |
+    +-------------+-------------------------------------+-------------------------+
+    | ``sleep``   | :c:macro:`CLOCK_MGMT_STATE_SLEEP`   | Clock state when        |
+    |             |                                     | the device is in low    |
+    |             |                                     | power or sleep modes    |
+    +-------------+-------------------------------------+-------------------------+
+
+Devicetree Representation
```

Comment:
Looks like this is out of sync with the actual dts changes and bindings you are introducing.

Try to figure out if you're suggesting we go away from `clocks` properties or not.